cmd/application.go

```diff
@@ -132,21 +133,40 @@ var applicationListCmd = &cobra.Command{
 	Use:   "list",
 	Short: "Lists all the applications",
 	Long:  "Lists all the applications",
-	Example: `  # List all applications
+	Example: `  # List all applications in current project
   odo app list
+
+  # List all applications in the specified project
+  odo app list --project myproject
 	`,
 	Args: cobra.ExactArgs(0),
 	Run: func(cmd *cobra.Command, args []string) {
 		client := getOcClient()
-		apps, err := application.List(client)
-		checkError(err, "")
-		fmt.Printf("ACTIVE   NAME\n")
-		for _, app := range apps {
-			activeMark := " "
-			if app.Active {
-				activeMark = "*"
+		if projectName == "" {
```

Comment:
I think it would be nice to have UTs for all validations..
But, the current practice is that we write UTs for sources in pkg/ but currently not for the cmd/ layer...
So, I feel it would be nice to only have the user facing messages only here but the rest of the things delegated to pkg layer...
 
Also, in the pkg/application/application.go#List I would suggest modifying it to take the form as under:

```

	if projectName != "" {
		exist, err := project.Exists(client, projectName)
		if err!= nil {
			return [], err// may be wrapped error to indicate context instead of just err
		}
		if !exist {
			return [], fmt.Errorf("Requested project does not exist")//Also suggest using odo project list to get list of projects
		}
	} else {
		projectName = project.GetCurrent(client)
	}
	....
	// Normal list code...
```

Reply:
where projectName is passed as an arguement to List function